### PR TITLE
Removes embedded Okta Groups

### PR DESCRIPTION
**Pull request by fgoncalves-io**
The current value for `local.okta_groups` is fixed and a mix of AWS and Github groups.

This change proposes separating AWS groups into a dedicate list and supporting custom groups in order to accommodate different needs. However, it still offers a default group `AWSPlatformAdmins` to avoid having to always pass a group name.

### CODE REVIEW

**Review by marwinbaumannsbp — APPROVED** [review]

**Comment by shoekstra** [comment]
I'd like to put this on hold until [support for SSO permission sets](https://github.com/hashicorp/terraform-provider-aws/pull/15322) has been merged until the AWS provider.

While it's just okta groups it doesn't make sense to be part of this module; if it's Okta groups together with permission sets then it makes sense.

**Comment by marwinbaumannsbp** [comment]
On the one hand I agree with you that its better to add support for Okta groups i.c.w. SSO permission sets so the full solution is implemented, but on the other hand not merging this makes the migration of the Geldmaat inception harder. If we are not going to merge this, are we then going to document that the creation of Okta groups is manual for now and remove this from the state? Maybe we should have a short call about this today! :-)

**Comment by fgoncalves-io** [comment]
We have decided to remove the Okta Groups from the module and manage them separately until we can implement the full solution including SSO permission set assignments.

File: variables.tf
Comment (shoekstra) [inline variables.tf:10]:
```suggestion
  description = "List of Okta group IDs that should be assigned the AWS SSO Okta app"
```

**Review by shoekstra — CHANGES_REQUESTED** [review]
Minor fix required 🙂

**Review by shoekstra — APPROVED** [review]

**Comment by burck1** [comment]
Hi @fgoncalves-io, @shoekstra, @marwinbaumannsbp. 

In the interest of getting the AWS SSO resources merged, please go give a thumbs up to [15808](https://github.com/hashicorp/terraform-provider-aws/pull/)

We've completed most of the work for supporting the [AWS SSO](https://docs.aws.amazon.com/singlesignon/latest/APIReference/welcome.html) and [AWS SSO Identity Store](https://docs.aws.amazon.com/singlesignon/latest/IdentityStoreAPIReference/welcome.html) resources and datasources in Terraform. The [15322](https://github.com/hashicorp/terraform-provider-aws/pull/) [WIP] PR encompasses all of that work. But, the contribution guide for the repo recommends submitting small pull requests with the minimum required resources, so we've submitted [15808](https://github.com/hashicorp/terraform-provider-aws/pull/) as our initial PR with just `data.aws_sso_instance`, `data.aws_sso_permission_set`, and `aws_sso_permission_set`. Once that's merged, we will submit PRs for all of the other resources and data sources since they depend on that initial PR.

**Pull request merged**
